mlos_core/mlos_core/optimizers/bayesian_optimizers.py

```diff
@@ -276,6 +277,177 @@ def surrogate_predict(self, configurations: pd.DataFrame, context: Optional[pd.D
         return ret
 
     def acquisition_function(self, configurations: pd.DataFrame,
-                             context: Optional[pd.DataFrame] = None) -> Callable:
+                             context: Optional[pd.DataFrame] = None) -> npt.NDArray:
         # This seems actually non-trivial to get out of skopt, so maybe we actually shouldn't implement this.
         raise NotImplementedError()
+
+
+class SmacOptimizer(BaseBayesianOptimizer):
+    """ Wrapper class for SMAC based Bayesian optimization.
+
+    Parameters
+    ----------
+    parameter_space : ConfigSpace.ConfigurationSpace
+        The parameter space to optimize.
+    """
+    def __init__(
+        self,
+        parameter_space: ConfigSpace.ConfigurationSpace,
+        space_adapter: Optional[BaseSpaceAdapter] = None,
+        seed: Optional[int] = None,
+        run_name: str = 'smac',
+        output_directory: str = 'smac3_output',
+        n_random_init: Optional[int] = 10,
+        n_random_probability: Optional[float] = 0.1,
+        n_workers: int = 1,
+    ):
+        super().__init__(parameter_space, space_adapter)
+
+        from smac import HyperparameterOptimizationFacade as Optimizer_Smac # pylint: disable=import-outside-toplevel
```

Comment:
Just write a pylint comment once on a preceding line. Also, make sure you have at least TWO spaces before each in-line comment (flake8 checks that)